Log unknown CME error code

maximeborges opened the pull request:
Currently when we receive an unknown error from the modem we don't really see any relevant info because every u-blox-specific error codes are mapped to unknown.
This PR print the error code so it can be matched in the [AT command manual](https://content.u-blox.com/sites/default/files/SARA-R5_ATCommands_UBX-19047455.pdf) (appending A1, p464)

andresv commented:
I think we should merge our changes to separate `yaak` branch and not to master. Otherwise it would be annoying to update our master with upstream.
And better yet let's try to merge it upstream.

andresv commented on atat/src/client.rs:
```diff
@@ -80,6 +80,39 @@ where
     }
 }
 
+/// Blocks on `nb` function calls but allow to time out
+/// Example of usage:
+/// `block_timeout!((client.timer, client.config.tx_timeout) => {client.tx.write(c)}.map_err(|_e| Error::Write))?;`
+/// `block_timeout!((client.timer, client.config.tx_timeout) => {client.tx.write(c)});`
+#[macro_export]
+macro_rules! block_timeout {
+    ($timer:expr, $duration:expr, $e:expr, $map_err:expr) => {{
+        if $duration == 0 {
+            nb::block!($e).map_err($map_err)
+        } else {
+            $timer.start($duration.millis()).ok();
+            loop {
+                match $e {
+                    Err(nb::Error::WouldBlock) => (),
+                    Err(nb::Error::Other(e)) => break Err($map_err(e)),
+                    Ok(r) => break Ok(r),
+                };
+                match $timer.wait() {
+                    Err(nb::Error::WouldBlock) => (),
+                    Err(_) => break Err(Error::Write),
+                    Ok(_) => break Err(Error::Timeout),
+                };
+            }
+        }
+    }};
+    (($timer:expr, $duration:expr) => {$e:expr}) => {{
+        block_timeout!($timer, $duration, $e, |e| { e })
+    }};
+    (($timer:expr, $duration:expr) => {$e:expr}.map_err($map_err:expr)) => {{
+        block_timeout!($timer, $duration, $e, $map_err)
+    }};
+}
+
```
This was already done right and github is showing it because our branching is what it is?

andresv replied:
Aa it is rebased on top of that https://github.com/yaak-ai/atat/pull/1

maximeborges commented:
I think there was an issue/conflict at some point with `embedded-hal` version, when they were updating to 0.9. Since we switched to `1.0.0-alpha.8` in the meantime, maybe it's worth trying to rebase our stuff on their `master`, then see if it's relevant to have our changed merge upsteram?

maximeborges commented:
I think there was an issue/conflict at some point with `embedded-hal` version, when they were updating to 0.9. Since we switched to `1.0.0-alpha.8` in the meantime, maybe it's worth trying to rebase our stuff on their `master`, then see if it's relevant to have our changed merge upstream?

andresv commented:
It is quite complicated. esp-idf-hal should be also updated to newer version which supports alpha-9, but it is not an easy task. I tried it once and then ran out of time.

I actually wanna do it, but not sure if it is possible within 2 upcoming sprints.

maximeborges commented:
Rhaa stupid cell networks in Germany, I saw my comment twice so I removed one but it deleted both :c  (edit: or apparently we deleted one each xD)
Also I was talking about `embedded-hal`0.9 and not 1.0.0-alpha.9.
They seem to have switched to `embedded-hal-nb` 1.0.0-alpha.1 now upstream so it might break even more things

maximeborges commented:
... okay this feature already got merged upstream actually
https://github.com/BlackbirdHQ/atat/pull/133
